### ISSUE TITLE
Approximately how many epoches will reach the results in the paper (4x SR result)

### ISSUE THREAD

**Issue by Senwang98** [liqilei/GMFN#7]
Hi, liqilei
After I have run about 700 epoches, the reult on val set is 32.41(highest result). I want to know if my training process seems to be problematic?
How long did you reach 32.47 of SRFBN when you were training? How long does it take to reach 32.70?
Thank you.

**Comment by liqilei** [comment]
Hi. For GMFN, each model is trained 1000 epochs. For SRFBN, you can refer to their paper for more details. 
Also, what do you mean by 32.70? Can you point out which model (I assume GMFN) and which scale (x4) on which dataset (I assume Set5) you are referring to ? Because the reported value is 32.55. For x4 on set 5.

**Comment by Senwang98** [comment]
Yes, you are right, GMFN final result is 32.55(Set5 4x). **Because it seems that your val set is the final test set(just take set5 into account)**, so I think the best result in training process is just that in final test set.
After **700** epoches, my best val set result is just **32.41**, do you think this is normal?(Because someone is using GPU, so I didnt finish the whole training process')
Or the final result will get in the last 200 epoches?(**800-1000th epoch**)?
I think you are Chinese, So if you can't get what I mean, we can talk in Chinese through email(my emial: egqawkq@gmail.com)

**Comment by liqilei** [comment]
So 32.70 is a typo?
I think this is abnormal because you may not get a 0.14 improvement in the final 300 epochs.

**Comment by Senwang98** [comment]
Yes, 32.7 is a typo.
This my json file, I think I didn't do any change to SRFBN.
```
{
    "mode": "sr",
    "use_cl": true,
    "gpu_ids": [0],

    "scale": 4,
    "is_train": true,
    "use_chop": true,
    "rgb_range": 255,
    "self_ensemble": false,
    "save_image": false,

    "datasets": {
        "train": {
            "mode": "LRHR",
            "dataroot_HR": "/home/wangsen/ws/dataset/DIV2K/Augment/DIV2K_train_HR_aug/x4",
            "dataroot_LR": "/home/wangsen/ws/dataset/DIV2K/Augment/DIV2K_train_LR_aug/x4",
            "data_type": "npy",
            "n_workers": 4,
            "batch_size": 64,
            "LR_size": 48,
            "use_flip": true,
            "use_rot": true,
            "noise": "."
        },
        "val": {
            "mode": "LRHR",
            "dataroot_HR": "./results/HR/Set5/x4",
            "dataroot_LR": "./results/LR/LRBI/Set5/x4",
            "data_type": "img"
        }
    },

    "networks": {
        "which_model": "GMFN",
        "num_features": 64,
        "in_channels": 3,
        "out_channels": 3,
        "num_steps": 2,
        "num_blocks": 7,
        "num_reroute_feats": 4,
        "num_refine_feats": 1
    },

    "solver": {
        "type": "ADAM",
        "learning_rate": 0.0002,
        "weight_decay": 0,
        "lr_scheme": "MultiStepLR",
        "lr_steps": [200, 400, 600, 800],
        "lr_gamma": 0.5,
        "loss_type": "l1",
        "manual_seed": 0,
        "num_epochs": 1000,
        "skip_threshold": 3,
        "split_batch": 1,
        "save_ckp_step": 50,
        "save_vis_step": 1,
        // "pretrain": null,
        "pretrain": "resume",
        "pretrained_path": "./experiments/GMFN_in3f64_x4/epochs/last_ckp.pth",
        "cl_weights": [1.0, 1.0, 1.0, 1.0]
    }
}
```

**Comment by Senwang98** [comment]
Okay, this time I am sure all file is unchanged except for batch_size and dataset path.
I will re-train from the scratch

**Comment by liqilei** [comment]
You settings seem OK. Please can you train a x2 model and share the performance? I remember the x3 and x4 models are fine-tuned based on x2 model.

**Comment by Senwang98** [comment]
Okay, I will re-train model(2x), please wait for several days(Because my 3090 GPU's cudnn version is 8.0.3 and it runs slower than 2080ti in CNN operation. This problem can be solved when the pytorch supports cudnn8.0.5)
![image](https://user-images.githubusercontent.com/23122139/100723234-6d42e900-33b9-11eb-8df1-39542e3e57e3.png)
